src/Microsoft.Data.Analysis/PrimitiveDataFrameColumnComputations.tt

```diff
@@ -112,7 +166,61 @@ namespace Microsoft.Data.Analysis
 <# } #>
             }
 <# } else if ((method.MethodName == "CumulativeMax" || method.MethodName == "CumulativeMin" || method.MethodName == "CumulativeProduct" || method.MethodName == "CumulativeSum") && method.SupportsRowSubsets == true) { #>
-            var ret = default(<#=type.TypeName#>);
+<# if (method.MethodName == "CumulativeMax") { #>
+    <# if (type.TypeName == "int") { #>
+        var ret = Int32.MinValue;
+    <# } else if (type.TypeName == "long") { #>
+        var ret = Int64.MinValue;
+    <# } else if (type.TypeName == "float") { #>
+        var ret = Single.MinValue;
+    <# } else if (type.TypeName == "decimal") { #>
+        var ret = Decimal.MinValue;
+    <# } else if (type.TypeName == "double") { #>
+        var ret = Double.MinValue;
+    <# } else if (type.TypeName == "short") { #>
+        var ret = Int16.MinValue;
+    <# } else if (type.TypeName == "byte") { #>
+        var ret = Byte.MinValue;
+    <# } else if (type.TypeName == "char") { #>
+        var ret = Char.MinValue;
+    <# } else if (type.TypeName == "sbyte") { #>
+        var ret = SByte.MinValue;
+    <# } else if (type.TypeName == "uint") { #>
+        var ret = UInt32.MinValue;
+    <# } else if (type.TypeName == "ulong") { #>
+        var ret = UInt64.MinValue;
+    <# } else if (type.TypeName == "ushort") { #>
+        var ret = UInt16.MinValue;
+    <# } #>
+<# } else if (method.MethodName == "CumulativeMin") { #>
+    <# if (type.TypeName == "int") { #>
+        var ret = Int32.MaxValue;
+    <# } else if (type.TypeName == "long") { #>
+        var ret = Int64.MaxValue;
+    <# } else if (type.TypeName == "float") { #>
+        var ret = Single.MaxValue;
+    <# } else if (type.TypeName == "decimal") { #>
+        var ret = Decimal.MaxValue;
+    <# } else if (type.TypeName == "double") { #>
+        var ret = Double.MaxValue;
+    <# } else if (type.TypeName == "short") { #>
+        var ret = Int16.MaxValue;
+    <# } else if (type.TypeName == "byte") { #>
+        var ret = Byte.MaxValue;
+    <# } else if (type.TypeName == "char") { #>
+        var ret = Char.MaxValue;
+    <# } else if (type.TypeName == "sbyte") { #>
+        var ret = SByte.MaxValue;
+    <# } else if (type.TypeName == "uint") { #>
+        var ret = UInt32.MaxValue;
+    <# } else if (type.TypeName == "ulong") { #>
+        var ret = UInt64.MaxValue;
+    <# } else if (type.TypeName == "ushort") { #>
+        var ret = UInt16.MaxValue;
+    <# } #>
+<# } else { #>
+    var ret = default(<#=type.TypeName#>);
```

Comment:
While you are changing the types here to `int`, `double`, `float` etc, can you also align the `var ret = ...` line so it lines up with the `var mutableBuffer = ...` line below? It's only a cosmetic change, but since you're going to go through these lines anyway, might as well change it.